### ISSUE TITLE
where is the code sfm

### ISSUE THREAD

**Issue by sunshineywz123** [xuelunshen/gim#3]

**Comment by sunshineywz123** [comment]
I am writing to inquire about the possibility of the SFM (Structure-from-Motion) code related to the gimdkm project being open-sourced. I have been following the progress of your work with great interest and believe that the release of the code could significantly benefit the research community.

As a researcher in the field, I am eager to explore the potential applications of your SFM technology. If the code is or will be made available, I would greatly appreciate any information regarding the timeline or conditions for its release.

Additionally, if there are any pre-release versions, beta tests, or documentation available for early access, I would be very interested in participating or obtaining access.

Thank you for your time and consideration. I look forward to any updates you can provide on this matter.

Best regards

**Comment by xuelunshen** [comment]
Thank you, @sunshineywz123 . The code for SfM is based on [hloc](https://github.com/cvg/Hierarchical-Localization). In detail, we have imitated the LoFTR in hloc to implement the SfM for DKM, because both of them can be regarded as detector-free matching network, so there are a lot of things to borrow from each other.

**Comment by truncs** [comment]
Thanks @xuelunshen 
What about the two-view matching? Is that opencv:triangulatePoints and converting that to PCD to visualize points?

For mutli view geometry with COLMAP, how many images were used for the reconstruction?

**Comment by xuelunshen** [comment]
@truncs 
The two-view matching is based on the [code](https://github.com/ubc-vision/COTR/blob/master/demo_reconstruction.py) of COTR. We replace the image matching part with different methods.

The multi-view reconstruction we used about 100+ images.

**Comment by terrance-wang** [comment]
Hi, thank you all for the amazing work on this project! I came across this project and was also interested in the application of your method with sfm for 3d reconstruction. I'm a bit new in this space though, and would really appreciate some help with hooking GIM into COLMAP and running some experiments with it. I've looked through the LoFTR implementation in hloc, but was wondering if I could get some more advice / help with this. I really want to replicate the results you guys posted on your website!

Thanks!
